specification/src/chapter3.adoc

```diff
@@ -50,6 +50,18 @@ yet].
 
 MMU, PMP/ePMP, and sPMP are discussed later in this chapter.
 
+==== PMA
+https://github.com/riscv/riscv-isa-manual/releases/tag/Priv-v1.12[Privileged
+ISA]
+
+_Physical memory attributes (PMA)_ are intended to capture inherent properties of the underlying hardware. For example, read-only ROM regions, or non-cachable device regions. Often PMA can be fixed at design time or at boot, but sometimes runtime PMA can be required.
+
+A separate hardware checker - _PMA checker_ - enforces PMA rules at runtime once a physical address is known. PMA rules are always checked on every physical access, and typically configured by region.
+
+Other RISC-V mechanisms can also be used for managing memory attributes. For example <<_mtt, MTT>> can provide finer grained dynamic PMA (svpam) at physical address level. Or <<_mmu, MMU>> TLB can be used by a kernel or hypervisor to manage memory attributes and permissions on a virtualized system.
+
+Which mechanism is used depends on use case. They can be combined, in which case any memory access must pass all memory attribute checks.
```

Comment:
what does it mean to pass memory attribute checks? PMA defines the memory behavior - perhaps you are mixing access-control checks with attributes?

Reply:
PMA is one mechanism for managing memory attributes (hardware attributes, at a granularity of a small(ish) number of regions). In practice defined at system reset.

Other mechanisms can also be used, including both better granularity and more dynamic, like svpam.

This whole area is a major weakness of RISC-V in my opinion. Other architectures do this at TLB level, with page granularity, with or without RME/GPT (MTT), with or without virtualization. Trying to achieve the same in RISC-V is ... challenging. 

Hence the discussion around PMA in other comments, for example. It might not be how the designers of RISC-V envisioned things like PMA being used, but that doesn't mean they can't be used in this way to work around shortcomings in the RISC-V architecture.


But that is a whole different discussion :-)

Reply:
the comment still remains - what does it mean to pass memory attribute checks?

Reply:
I deleted the offending paragraph.